### PR TITLE
Reorganize documentation as per Diataxis framework

### PR DESCRIPTION
**Pull request by aldesantis**
This is a first attempt at reorganizing the documentation according to the [Diátaxis](https://diataxis.fr/) framework, which we want to start following to organize these docs.

Because of the sheer amount of content, many areas are still somewhat incomplete and/or not fully compliant with Diátaxis. 

As a next step, we'll establish a set of guidelines and best practices to follow (in the form of a repo readme and PR template), and then we'll start to apply "opportunistic refactoring" to the documentation to bring it closer to its ideal state.

### CODE REVIEW

**Comment by waiting-for-dev** [comment]
Hey @aldesantis! I think that adopting Diátaxis is a great idea and that it'll help a lot both users in their expectations and authors in their approach to making it evolve.

I'm only concerned that approaching it in a big first restructuration might not be the best idea. In fact, [Diátaxis warns against it](https://diataxis.fr/how-to-use-diataxis/#work-one-step-at-a-time):

> It’s natural to want to complete large tranches of work before you publish them, so that you have something substantial to show each time. Avoid this temptation - every step in the right direction is worth publishing immediately.
>
> Although Diátaxis is intended to provide a big picture of documentation, don’t try to work on the big picture. It’s both unnecessary and unhelpful. Diátaxis is designed to guide small steps; keep taking small steps to arrive where you want to go.

Don't get me wrong. I think there's a lot of value in your work, but we should add it incrementally. For instance, I believe you correctly noticed that most of our documentation consists of how-to's. In the first iterations, we could start detecting and extracting them, one by one. If we do all of them at once, it's easy to forget some needed care to make the whole picture consistent. I don't mean that we should make them 100% Diátaxis compliant on the first extraction, but, for instance, I saw that some have a very abrupt beginning, and some links are broken. Besides, it isn't straightforward to review from the outside.

Trying to reach the Diátaxis structure too quickly might even be harmful. The documentation we have now is not perfect, but there's more or less a consistent approach. If we make mention of the framework but don't implement it, it might be confusing. 

We've tried to introduce the four types of documentation, but I don't think that what we have put within `References` are actual references. I think they're a mix between explanation and how-to.

The same goes for the "Launching your first store" tutorial. In my view, all the "Customize" sub-sections are also a mix between explanations and how-tos. They lack a shared narrative; users don't go to a well-defined end, so I think that's very far from Diátaxis's view for a tutorial.

To sum up, this PR does a great job identifying what steps we need to pursue. But in my view, we should take them one by one, slowly. We can start extracting how-to's or polishing the extension's tutorial to make it even more tutorial*ish*, as it's much closer than the "Launching your first store" one.

What are your thoughts? :slightly_smiling_face:

**Comment by aldesantis** [comment]
@waiting-for-dev I think you're right. My idea was to do this initial pass, which leaves things in a somewhat working state, and then try to "opportunistically" fix everything else. However, re-reading the docs after this refactoring, we now have a bunch of guides that are harder to follow than before.

How about this as an alternative action plan?

- First, update the readme and add the PR template, so that any new guides are written according to Diataxis.
- Then, start forming the different sections (e.g., Tutorials, How-Tos), proceeding one small piece at a time (maybe just do one Solidus module at a time).
- Finally, update the main page to talk about the structure of the documentation and how to use it.

**Comment by waiting-for-dev** [comment]
Hey @aldesantis, I overall agree with your plan:

> Then, start forming the different sections (e.g., Tutorials, How-Tos), proceeding one small piece at a time (maybe just do one Solidus module at a time).

I've some concerns about taking this approach from the outside, though. As per **[Diátaxis recommendations](https://diataxis.fr/how-to-use-diataxis/#don-t-worry-about-structure)** (emphasis is mine):

> Although structure is key to documentation, using **Diátaxis means not spending energy trying to get its structure correct**.
>
> If you continue to follow the prompts that Diátaxis provides, eventually your documentation will assume the Diátaxis structure - but it will have assumed that shape because it has been improved. **It’s not the other way round, that the structure must be imposed upon documentation to improve it**.
>
> In practice, this means that getting started with **Diátaxis doesn’t require thinking about dividing up your documentation into the four sections**, or writing out headings to put material under.
>
> Instead, following the workflow described in the next two sections, make changes where you see opportunities for improvement according to Diátaxis principles, so that the **documentation starts to take a certain shape**. At a certain point, the changes you have made **will appear to demand that you move material under a certain Diátaxis heading** - and that is how your top-level structure will form. In other words, Diátaxis changes the structure of your documentation from the inside.

Instead of taking the four Diátaxis sections as a starting point, they should be our guide and, eventually, the ending point. We should start with the current content and re-shape it in a more Diátaxis-kind structure. I can see that the vast majority of our pages can now be broken down into Explanation and How-tos, and maybe we can also easily extract one or a couple of tutorials. But I wouldn't enforce the structure upfront. As they recommend, I'd add a new How-tos heading where we could add some of the extractions while gradually leaving the current structure as an Explanation "by default". Maybe we can nest the how-tos within the corresponding section. Eventually, we can adapt the recommended titles when there's nothing else to extract. WDYT?

**Comment by kennyadsl** [comment]
I agree with @waiting-for-dev  (and Diàtaxis docs) that we shouldn't have done this upfront but at this point what is done is done. At this point, we should think if it's better to keep optimizing the structure starting from what we have now or from what Alessandro made. 

To be honest, if we want to go in this direction, this PR looks like a better starting point to me. Maybe we didn't go to this point in the right way, and we still need to improve a lot, but it's likely more similar to what we want to see in the guides.

**Comment by aldesantis** [comment]
We might want to try something in-between: e.g., we could start by extracting the tutorials (which are already in almost-perfect shape, according to Diataxis guidelines), and take it from there, one piece at a time.

Let me reflect on this once I'm back from vacation next week and take another pass at it! 🙂

**Comment by waiting-for-dev** [comment]
Thanks, @aldesantis. Let us know what you come up with. In the meantime, I've leveraged https://github.com/solidusio/edgeguides/pull/61 to make a POC about how I envision it. Please, let me know your thoughts!

**Comment by aldesantis** [comment]
@waiting-for-dev let's go with your branch as a starting point. 🙂
